types/index.d.ts

```diff
@@ -76,6 +76,9 @@ declare module '@azure/functions' {
          * @param err A user-defined error to pass back to the runtime. If present, your function execution will fail.
          * @param result An object containing output binding data. `result` will be passed to JSON.stringify unless it is
          *  a string, Buffer, ArrayBufferView, or number.
+         *
+         * @deprecated Use of sync functions with `context.done()` is not recommended. Use async/await and pass the response as the return value instead.
```

Comment:
> Question: is this considered a breaking change? If a customer was also using the deprecation linting flag, this would break previously working code.

No, the point of deprecating something is to give users a heads up _before_ you make a breaking change. We currently have no intention of removing `context.done`, but marking it as deprecated now gives users _plenty_ of time to change their code in case we ever decide to remove it.

Also, lint rules are purely optional and we don't consider them for breaking changes.

Reply:
Great, thank you!